### PR TITLE
[e2e test]add telemetry e2e test with default provider

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

add telemetry e2e test with default provider

### CODE REVIEW

**Review by douglas-reid — COMMENTED** [review]
Can we move the access logging tests out of the `stats/prometheus` hierarchy?

i feel like this should be something like: `tests/integration/telemetry/accesslogs`.

@howardjohn how did this end up inside the `stats` hierarchy again?  were we trying to save on setup costs?  do we need to consider a refactor that exercises all telemetry API features instead (`tests/integration/telemetry/api`) ?

**Comment by zirain** [comment]
> 

+1 move all accesslogs e2e tests to an new folder.

**Comment by zirain** [comment]
open a [ticket](https://github.com/istio/istio/issues/39623) to track this, cc @douglas-reid

File: tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go [hunk 1]
```diff
@@ -51,15 +52,55 @@ func TestAccessLogs(t *testing.T) {
...
 }
 
-func runAccessLogsTests(t framework.TestContext, expectLogs bool) {
+func TestAccessLogsDefaultProvider(t *testing.T) {
+	framework.NewTest(t).
+		Features("observability.telemetry.logging.defaultprovider").
+		Run(func(t framework.TestContext) {
+			t.NewSubTest("disabled").Run(func(t framework.TestContext) {
+				cfg := `
+accessLogFile: "/dev/null"
```
Comment (howardjohn) [inline tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go:73]: a bit confused about /dev/null.. should it be `""` if we are testing disabled?

File: tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go [hunk 2]
```diff
@@ -51,15 +52,55 @@ func TestAccessLogs(t *testing.T) {
...
+`
+				ist := *(common.GetIstioInstance())
+				istio.PatchMeshConfigOrFail(t, ist.Settings().SystemNamespace, t.Clusters(), cfg)
+				runAccessLogsTests(t, true)
+				cfg = `
+accessLogFile: "/dev/stdout"
+defaultProviders:
+  accessLogging: []
+`
+				istio.PatchMeshConfigOrFail(t, ist.Settings().SystemNamespace, t.Clusters(), cfg)
```
Comment (howardjohn) [inline tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go:98]: what is this doing? If its trying to revert the change,PatchMeshConfigOrFail automatically does that?

Comment (zirain) [reply tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go:98]: you're right, the code is move from a really branch, in prev context there's no `PatchMeshConfigOrFail`, we do not need it now.

File: tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go (hunk 1 above)
Comment (zirain) [reply tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go]: fixed

File: tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go (hunk 2 above)
Comment (zirain) [reply tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go]: fixed

File: tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go [hunk 3]
```diff
@@ -69,13 +99,30 @@ spec:
   - providers:
     - name: envoy
     disabled: %v
-`, !expectLogs)
+`, !enableLogs)
+	t.ConfigIstio().YAML(common.GetAppNamespace().Name(), config).ApplyOrFail(t)
+}
+
+func deleteTelemetryResource(t framework.TestContext, enableLogs bool) {
```
Comment (douglas-reid) [inline tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go:106]: I'm confused. This looks exactly the same as `applyTelemetryResource`. Shouldn't this end with a `DeleteOrFail` ?

**Review by douglas-reid — APPROVED** [review]
